src/components/Products.js

```diff
@@ -1,14 +1,23 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import ProductCard from './ProductCard';
 import ProductsData from '../mocks/en-us/featured-products.json';
 import { ProductList } from './Products.styles';
+import Pagination from '../components/Pagination';
+
+export default function Products({ categoriesSelected }) {
+  let [products, setProducts] = useState([]);
+  useEffect(() => {
+    let mounted = true;
+    const productsData =
+      categoriesSelected && categoriesSelected.length > 0
+        ? ProductsData.results.filter((prod) =>
+            categoriesSelected.includes(prod.data.category.id)
+          )
+        : ProductsData.results;
+    if (mounted) setProducts(productsData);
+    return () => (mounted = false);
```

Comment:
Para la variable `mounted` que tiene recomendaría que usaras el hook `useRef`, porque la implementación que tienes aquí ahorita aún podría resultar en algunos casos donde se intente hacer el set sin el componente estar montado.

Quedaría algo así.

```suggestion
    const mounted = useRef(true);
    const productsData =
      categoriesSelected && categoriesSelected.length > 0
        ? ProductsData.results.filter((prod) =>
            categoriesSelected.includes(prod.data.category.id)
          )
        : ProductsData.results;
    if (mounted.current) setProducts(productsData);
    return () => (mounted.current = false);
```

De esta forma la referencia persiste a pesar de los re renders que pueda tener el component. La idea es la misma, solo aplicamos la solución más estándar para este tipo de problemas.

Reply:
También creo que toda la lógica para obtener productsData bien podría estar en una función en un archivo `utils.js` para que el código te quede nada más así:

```suggestion
    let mounted = true;
    const filteredProducts = filterProducts( ProductsData );
    if (mounted) setProducts(productsData);
    return () => (mounted = false);
```

Así le quitamos lógica de negocio al componente y mantenemos solo lógica de render en él. Además, gana mucha legibilidad.

Reply:
se agrega custom hook